tests/min_mapped_reads.nf.test

```diff
@@ -13,797 +13,17 @@ nextflow_pipeline {
-                    file("${params.outdir}/star_salmon/RAP1_UNINDUCED_REP2/quant.genes.sf").name,
-                    file("${params.outdir}/star_salmon/RAP1_UNINDUCED_REP2/quant.sf").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/aux_info/fld.gz").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/aux_info/meta_info.json").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/libParams/flenDist.txt").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/logs/salmon_quant.log").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/quant.genes.sf").name,
-                    file("${params.outdir}/star_salmon/WT_REP1/quant.sf").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/aux_info/fld.gz").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/aux_info/meta_info.json").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/libParams/flenDist.txt").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/logs/salmon_quant.log").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/quant.genes.sf").name,
-                    file("${params.outdir}/star_salmon/WT_REP2/quant.sf").name
-                ).match("star_salmon/salmon_quant") },
-                { assert snapshot(
-                    // These reports are not stable
-                    file("${params.outdir}/trimgalore/RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP1_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP1_trimmed_2.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP2_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP2_trimmed_2.fastq.gz_trimming_report.txt").name
-                ).match("trimgalore") },
-                { assert snapshot(
-                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml")
-                ).match("versions") }
+                    workflow.trace.tasks().size(),
+                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
+                    file_complement.collect{ file -> file_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
+                    file_snapshottable
```

Comment:
```suggestion
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    file_complement.collect(),
                    file_snapshottable
```

Then if the file complement contains relative paths rather than names, and we don't exclude the snapshotted files (I still don't think we need to), then we can just do something like this, right?